javagen/src/main/resources/OperationLocationPollingStrategy.java

```diff
@@ -0,0 +1,127 @@
+import com.azure.core.exception.AzureException;
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.OperationResourcePollingStrategy;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollingContext;
+import com.azure.core.util.polling.PollingStrategyOptions;
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+
+/**
+ * Implements an operation location polling strategy, from Operation-Location.
+ *
+ * @param <T> the type of the response type from a polling call, or BinaryData if raw response body should be kept
+ * @param <U> the type of the final result object to deserialize into, or BinaryData if raw response body should be
+ * kept
+ */
+public final class OperationLocationPollingStrategy<T, U> extends OperationResourcePollingStrategy<T, U> {
+
+    private static final ClientLogger LOGGER = new ClientLogger(OperationLocationPollingStrategy.class);
+
+    private final ObjectSerializer serializer;
+    private final String endpoint;
+    private final String propertyName;
+
+    /**
+     * Creates an instance of the operation resource polling strategy.
+     *
+     * @param pollingStrategyOptions options to configure this polling strategy.
+     * @throws NullPointerException if {@code pollingStrategyOptions} is null.
+     */
+    public OperationLocationPollingStrategy(PollingStrategyOptions pollingStrategyOptions) {
+        this(pollingStrategyOptions, null);
+    }
```

Comment:
In this mode, it works almost same as azure-core `OperationResourcePollingStrategy`, except that it would not throw if the initial response cannot be de-serialize to `T` (as in PUT pattern, it could be `U`)